### ISSUE TITLE
[New build - DEFAULT] 22w12a: The Fuel Saga #4: A New Harvest

### ISSUE THREAD

**Issue by tsunamayo** [tsunamayo/Starship-EVO#4647]
A new fuel harvesting gameplay mechanic has been added.
Gas clouds will spawn close to the sun and are harvestable by the player for refuel.
To do do link (with pipes of course) a Fuel Scoop to a tank and fly inside the gas cloud.

Gas Cloud Feature:
- Expand or retract depending on a much fuel is stored inside the cloud.
- Built using the latest particle system technology.
- Will be visible far away through a LOD system.
- Largest cloud can be selected as tracking target, so you can check how much fuel is left.

Fuel Scoop:
- Can be set to harvest or Expel in case you need to dump fuel.
- In case of expel a new gas cloud will be created if necessary.
- Can be toggled On or Off.
- It has three signal events for all your logic needs : [Is harvesting], [Is Expelling] and [Is Inside Gas Cloud]
- Has a harvest and expel particle effect.
- All built using DOTS technology, so it should be super fast!

Other features:
- HUD Fuel fuel widget has been improved. 

WIP:
- Fuel rate will likely require more tuning as it might be too slow.
- The gas clouds pops-in a bit abruptly. This will be fixed once the size of the world will be reduced (that will be necessary for other reasons too).

**Comment by ProPeach** [comment]
Yahoo! The Scoops having Event Gate compatibility right from the start is awesome!

This DOTS tech also seems super powerful, adding features for almost no performance cost almost feels like cheating😆

**Comment by pinesh** [comment]
A reduction in world size? Does this have anything to do with the unity distance precision problem or is this a creative choice?

**Comment by Drillz007** [comment]
Will gas clouds refill over time or spawn randomly? or will it be a more rare occurrence so players still buy gas from stations if they use up all the local methane clouds?

**Comment by ZachZent** [comment]
Since fuel chemistry has been a discussion since the kickstarter days, do you have any updates on your plans now that we actually have fuel to chemistry? 

I'm personally in favor of the base fuel being functional with its balance statistics. Combined fuel and minerals in a centrifuge, which must spin as it's the law, to create different fuels with varying statistics. One is faster but has a high use rate sort of thing.

**Comment by tsunamayo** [comment]
Build is up!
@ProPeach yep they are quite quick to add if it is something you get naturally in the code
@pinesh no we are still well over normal floating point issues. It is for game design, also it will be less stress on some part of the tech.
@Drillz007 They will respawn if you move back, and you wont run out of fuel they are too big.

Btw I have added a few question to the steam post if you want to discuss there also (about multi-fuel).

**Comment by tsunamayo** [comment]
> Since fuel chemistry has been a discussion since the kickstarter days, do you have any updates on your plans now that we actually have fuel to chemistry?
> 
> I'm personally in favor of the base fuel being functional with its balance statistics. Combined fuel and minerals in a centrifuge, which must spin as it's the law, to create different fuels with varying statistics. One is faster but has a high use rate sort of thing.

Yeah I have asked the question on the steam post to get feedback from other people too, you can contribute there too if you want. I am just trying a bit to see if I can engage other people too (but you can contribute here if you want).
I would like multi-fuel but I fear it will be a pain:
say you found a better fuel: now you need to unload your ship fuel, reconfigure your fuel scoop and refill. Sounds a bit too much, although harvesting the higher fuel wont be mandatory.

**Comment by Womble-TC** [comment]
Love the clouds! Two things:

-Do you plan on making the clouds vary slightly in colour, perhaps depending on the star they spawn around?
-In future, will sheilds be effected when entering gas clouds? Could be an interesting mechanic for combat scenarios in these areas.

**Comment by ZachZent** [comment]
`Yeah I have asked the question on the steam post to get feedback from other people too, you can contribute there too if you want. I am just trying a bit to see if I can engage other people too (but you can contribute here if you want).
I would like multi-fuel but I fear it will be a pain:
say you found a better fuel: now you need to unload your ship fuel, reconfigure your fuel scoop and refill. Sounds a bit too much, although harvesting the higher fuel wont be mandatory`

I can give you two options

1. Tiers, no bullshit
Tiers are completely technologically separated. Class II fuel can only be used on a Class II engine which can only be stored in tanks with Class II fuel. Class II fuel can only be found in more risky areas, say closer to the sun where there is more risk of heat damage or whatever else. Not rare, but you need to get the right hardware to get the better stuff. Sort of like avorion

2. Ditch tiers
I want chemistry, but any idea I come up with would make it just too complicated. See below, "the bottle". Keep it at one fuel type for cruise. Clouds have different densities in riskier areas, again closer to the sun which makes extraction both for a ship and a station faster.

For both of them you can have refined and unrefined fuels, which should absolutely be a thing. It's another gameplay path and an economy on its own.


For fuel chemistry and variety, I'd stick to keeping that with warp fuel. I remember you saying that you liked the NMS model of warp fuel in principle. Many different items which you unlock as you progress to create fuel. If you keep it that sort of "plug and play" model, regardless if it is one item per jump or percentages of the core per jump, you can add fuel chemistry to change the statistics of the "fuel rods"



![image](https://user-images.githubusercontent.com/36980446/160037041-52644d2f-872a-4aa2-a85e-c7470135188f.png)

**Comment by Hecateus** [comment]
iirc, there is an interest in fuels alternative to Gas-Cloud Fuel.  OK;

Basic Hydro-fuels: Methane and other hydrocarbon fuels, such as Ammonia, Magnesium-hydride, even hot steam-water.  High thrust, Low Efficiency, used by low-tech spacefarers.  Found most everywhere in the galaxy.

Fission Fuels: medium efficiency and high thrust; also requires suitable heat-able propellant.  Also used by low tech spacefarers with a good access to fissile material and haven't quite mastered compact fusion energy.

Anti-Matter: (aka Mirror Matter)  High Efficiency and High Thrust, but very expensive, and usually used as a War Emergency Power added to souped up engines.  Can be slowly collected at dedicated orbital stations around gas giants (created by lightning storms).

**Comment by ProPeach** [comment]
Agreed on the slow rate of harvesting fuel, this Mistral test ship has 4x1m scoops on itand consumes fuel at exactly the rate it is harvesting at. Looks like it needs at least a 5-10 times harvesting rate buff!
![image](https://user-images.githubusercontent.com/27857711/160262190-e8520aaf-9444-40a1-8dde-f9a097b93cee.png)
[Mistral - Fuel Test.zip](https://github.com/tsunamayo/Starship-EVO/files/8356615/Mistral.-.Fuel.Test.zip)

**Comment by Horkas** [comment]
I agree with @ZachZent 's viewpoint of using tiers. Adding any more complexity to the game will negatively effect it unless you are going for a more hard core survival aspect for the game. Keep it simple and tier based.

- Having fuel alone as a mechanic to manage will be too big of a big pain for most casual players to deal with, (Your target audience for maximum profit potential.) Adding even more complexity will tank the games possible popularity the way of Starmade and its reactor shenanigan's or From the Depths and its entire building systems.

- Keep in mind, that the more complex non-direct gameplay is, (Aka, dealing with ship logistics when building.), the more off putting it is for an average player to want to pick up and play the game. Complexity is good! But you must understand that it will cut your potential players/supports into more and more niche of an audience. So you are sort of shooting yourself in the foot by adding all of this building complexity if your goal is to sell allot of copies of the game. (This is not random bs I'm making up. This is derived from hundreds of hours of focus testing for various games.)

- Fueling should not be a chore in any way shape or form. Managing it and making sure you don't consume it too quickly is chore enough, but having to spend long amounts of time sucking up the clouds? Forget it.

- Under no circumstances should the game ever feel "Tedious" to play. So when making new content or balancing it, always pay attention to if you actively hate or get aggravated by something. If you do, it needs changed. (There are ways to circumvent tedious gameplay, by adding small rewards like exp drops or currency for certain actions, but this should be used only as a last resort if the content just cannot be made fun.)

- Try to remember, that based on your games art style, your target audience is going to tend to be younger. Age ranges 9-14. No matter how you market it, kids will see pretty colors and ask their parents to buy the "Neat space game" for them. Try your best to allow players to get right into the game, no building needed, and start exploring the galaxy, once it gets to that stage. (Think how big minecraft/Fortnite is in the childrens pop culture for reference.) If a kid can't play the game, it needs tuned. (Again, based on focus testing for ALLOT of games. You would be ....... disheartened to see how similar adults are to kids in regards to mentally understanding games and how to play them. The gap is not large.)
